kubectl-volsync/cmd/migration_create.go

```diff
@@ -74,24 +123,163 @@ func doMigrationCreate(cmd *cobra.Command, args []string) error {
+	}
+
+	if destPVCExists {
+		rsyncSpec.DestinationPVC = &migParams.Dest.PVC
+	}
+
+	rd := &volsyncv1alpha1.ReplicationDestination{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "volsync.backube/v1alpha1",
+			Kind:       "ReplicationDestination",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "migration-destination",
+			Namespace: migParams.Dest.Namespace,
+		},
+		Spec: volsyncv1alpha1.ReplicationDestinationSpec{
+			Rsync: rsyncSpec,
+		},
+	}
+	klog.V(2).Infof("Creating ReplicationDestination in namespace %s", migParams.Dest.Namespace)
+	if err := migParams.Dest.Opts.Client.Create(ctx, rd); err != nil {
+		return err
+	}
+	klog.V(0).Infof("ReplicationDestination created in namespace %s", migParams.Dest.Namespace)
+
+	return nil
+}
+
+func prepMigrationParamsStruct(cmd *cobra.Command, relation *Relationship) (*MigrationParams, error) {
 	// Insert information into the relationship & save it
 	cap, err := cmd.Flags().GetString("capacity")
 	if err != nil {
-		return err
+		return nil, err
 	}
 	relation.Set("capacity", cap)
 
+	nsName, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("namespace", nsName)
+
+	pvcName, err := cmd.Flags().GetString("pvcname")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("pvcname", pvcName)
+
+	accessMode, err := cmd.Flags().GetString("accessmodes")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("accessmodes", accessMode)
+
+	serviceType, err := cmd.Flags().GetString("servicetype")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("serviceType", serviceType)
 	if err = relation.Save(); err != nil {
-		return fmt.Errorf("unable to save relationship configuration: %w", err)
+		return nil, fmt.Errorf("unable to save relationship configuration: %w", err)
 	}
 
 	// TODO: Set up objects on the cluster
-	_, _ = newClient("")
+	clientObject, _ := newClient("")
+
+	return &MigrationParams{
+		Dest: DestinationInfo{
+			Namespace:   nsName,
+			PVC:         pvcName,
+			Capacity:    cap,
+			ServiceType: serviceType,
+			Opts: VolSyncOptions{
+				Client: clientObject,
+			},
+		},
+	}, nil
+}
 
+//nolint:funlen
+func GetDestinationPVC(ctx context.Context, migParams *MigrationParams) (*corev1.PersistentVolumeClaim, error) {
+	destPVC := &corev1.PersistentVolumeClaim{}
+	pvcInfo := types.NamespacedName{
+		Namespace: migParams.Dest.Namespace,
+		Name:      migParams.Dest.PVC,
+	}
+
+	err := migParams.Dest.Opts.Client.Get(ctx, pvcInfo, destPVC)
+	if err != nil {
+		return nil, err
+	}
+
+	return destPVC, nil
+}
+
+func checkAndCreateNamespace(ctx context.Context, migParams *MigrationParams) error {
+	ns := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: migParams.Dest.Namespace,
+		},
+	}
+
+	klog.V(2).Infof("Creating NameSpace %s ", migParams.Dest.Namespace)
+	if err := migParams.Dest.Opts.Client.Create(ctx, ns); err != nil {
```

Comment:
Filter out the error case where the NS exists here (`kerrs.IsAlreadyExists(err)`) before returning.